### PR TITLE
[GSOC 2.2] Add method to evaluate system response of Transfer Functions. 

### LINKED ISSUE #25315: How should `subs()` be handled for Transfer Functions
Right now, `.subs()` does not work with `TransferFunction` and `TransferFunction` Matrix class.
I think it should be supported as it is common to evaluate an input transfer function at a particular frequency `jw`.

Some control packages support the direct substitution:
**MATLAB**
```
>> t = tf([1 3],[1 1 1 1])
>> value = evalfr(t, 2);
>> value
    0.3333
```
**Python-control**
```
>>> tf = control.TransferFunction([1,3], [1,1,1,1])
>>> substituted_tf = control.evalfr(tf, 2)               # Same name, a coincidence :)
>>> substituted_tf
(0.3333333333333333+0j)
```

Relevant Comment :
1. https://github.com/sympy/sympy/pull/19390#issuecomment-640556483

### PR DESCRIPTION
**Pull request by faze-geek**
#### References to other Issues or PRs

Fixes #25315 

#### Release Notes

* physics.control
  * Added evalfr method to TransferFunction and TransferFunctionMatrix class.

### CODE REVIEW

**Review by Sc0rpi0n101 — APPROVED** [review]
LGTM

**Comment by moorepants** [comment]
Why is there a special method needed for this? Shouldn't .subs() or .xreplace() be able to replace s with anything you want?

File: sympy/physics/control/lti.py [hunk 1]
```diff
@@ -818,6 +818,32 @@ def zeros(self):
...
+        Examples
+        ========
+
+        >>> from sympy.abc import s, p, a
+        >>> from sympy.physics.control.lti import TransferFunction
+        >>> from sympy import I
+        >>> tf1 = TransferFunction(1, s**2 + 2*s + 1, s)
+        >>> omega = 0.1
+        >>> tf1.evalfr(I*omega)
+        TransferFunction(1, 0.99 + 0.2*I, s)
```
Comment (oscarbenjamin) [inline sympy/physics/control/lti.py:834]: It does not really seem meaningful to call this a transfer function.

**Comment by oscarbenjamin** [comment]
> Why is there a special method needed for this? Shouldn't .subs() or .xreplace() be able to replace s with anything you want?

I would expect `subs` to change the transfer function variable e.g.:
```python
>>> TransferFunction(1, s**2 + 2*s + 1, s).subs(s, t)
TransferFunction(1, t**2 + 2*t + 1, t) # variable is t
```
The method here returns
```python
>>> TransferFunction(1, s**2 + 2*s + 1, s).evalfr(t)
TransferFunction(1, t**2 + 2*t + 1, s) # variable is still s
```
So the s is replaced by t in the numerator and denominator but the function variable is still s.

I don't know much about control theory but it seems like the output here should be Expr rather than a TransferFunction because when you call/evaluate a function what is returned is not a function any more. Maybe it would make sense to use `__call__` and return an ordinary expression:
```python
>>> tf = TransferFunction(1, s**2 + 2*s + 1, s)
>>> tf(t)
1/(t**2 + 2*t + 1)
```

**Comment by moorepants** [comment]
I think this PR needs to be reverted. It needs some more thinking and alignment with how SymPy objects work.

**Comment by faze-geek** [comment]
The above points make sense.

1. To address the name of the function let's keep `evalfr` in full form as `eval_frequency` (or maybe even `evaluate_frequency`) so that there is no confusion for users. `evalfr` is the standard function in MALTAB and python-control so I went ahead with the name but it is clashing with Sympy.
2. We will return an expression for `TransferFunction` and a matrix for `TransferFunctionMatrix`.


This diff would do the job - 
```
-    def evalfr(self,other):
+    def eval_frequency(self,other):
         """
         Returns the system response at any point in the real or complex plane.

@@ -841,7 +841,7 @@ def eval_frequency(self,other):
         """
-        argnew = TransferFunction(arg_num, arg_den, self.var)
+        argnew = TransferFunction(arg_num, arg_den, self.var).to_expr()
```

Now -
```
>>> tf1 = TransferFunction(1, s**2 + 2*s + 1, s)
>>> omega = 0.1
>>> tf1.eval_frequency(I*omega)
1/(0.99 + 0.2*I)
>>> tf2 = TransferFunction(s**2, a*s + p, s)
>>> tf2.eval_frequency(2)
4/(2*a + p)
>>> tf2.eval_frequency(I*2)
-4/(2*I*a + p)
```

3. For `.subs()` kindly go through this [discussion](https://github.com/sympy/sympy/pull/19390#issuecomment-640556483) on why subs shouldn't work with TransferFunction.

**Comment by moorepants** [comment]
> evalfr is the standard function in MALTAB and python-control so I went ahead with the name but it is clashing with Sympy.

I don't recommend copying MATLAB function names, as they are cryptic and not Pythonic. Please use informative Python style names for things.

**Comment by moorepants** [comment]
I no longer see a "revert" button. I am fairly certain that I used to be able to see that and press it. We need to revert this. @faze-geek please make a new PR that reverts this.

**Comment by moorepants** [comment]
Please open a new issue or PR to discuss how this functionality should be designed. There are two possible things to return: 1) a function that can be evaluated at any frequency or 2) the result of evaluation at a specific frequency. Providing 1) then allows the user to evaluate that expression afterwards using any normal SymPy methods.

**Comment by oscarbenjamin** [comment]
> I no longer see a "revert" button.

The revert button is above on the note showing that the commit was merged to master (Ctrl-f "revert").

**Comment by moorepants** [comment]
You may see it, but it is not visible for me.

**Comment by oscarbenjamin** [comment]
> You may see it, but it is not visible for me.

Okay, I wonder why that is.

I've clicked it to open gh-25409.

**Pull request merged**